src/stan-users-guide/latent-discrete.Rmd

```diff
@@ -1019,3 +1019,182 @@ the probability mass function of $z_i$ in the posterior.   These may
+from the joint posterior $p_{X, Y}(x, y)$ and then estimate $\mu$ with
+
+$$
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M {g(x_i, y_i)}.
+$$
+
+If $Y$ is discrete, however, we cannot use Stan to sample from the joint 
+posterior because Stan cannot compute the derivatives it needs for the 
+discrete parameters. 
+
+Instead, as outlined in the previous sections, we first marginalize out $Y$ 
+from the joint posterior to give the marginalized posterior $p_X(x)$. This
+marginalized posterior can then be implemented in Stan as usual, and Stan will 
+give samples $\{x_i\}_{i = 1}^M$ from the marginalized posterior. 
+
+How, using only the samples $\{x_i\}_{i = 1}^M$, can we estimate 
+$\textsf{E}(g(X, Y))$? We can use a conditional estimator. That is, for each
+function $g$ of interest we can compute
+
+$$
+h(X) = \textsf{E}(g(X, Y) \mid X)
+$$
+
+and then estimate $\textsf{E}(g(X, Y))$ with
+
+$$
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M h(x_i).
+$$
+
+This estimator is justified by the fact that
+
+$$
+\textsf{E}(h(X)) 
+= \textsf{E}(\textsf{E}(g(X, Y) \mid X)) 
+= \textsf{E}(g(X, Y)) = \mu,
+$$
+
+which uses the [law of total expectation](https://en.wikipedia.org/wiki/Law_of_total_expectation). Using 
+this marginalized estimator gives us a way to estimate the expectation of any
+function $g$ for any combinations of discrete or continuous parameters in the 
+model. However, it presents a new challenge: evaluating the conditional 
+expectation $\textsf{E}(g(X, Y) \mid X)$.
+
+### Recovering the continuous parameters {-}
+
+In general, evaluating the expectation can be difficult, especially if we are
+interested in functions of both the discrete and continuous parameters. 
+Luckily, in practice we are commonly interested in expectations involving only 
+one of either the continuous or discrete parameters, for example the posterior
+mean of an individual scalar component of these parameters. In these cases,
+evaluating $\textsf{E}(g(X, Y) \mid X)$ becomes tractable.
+
+If $g(X, Y)$ is a function of only the continuous parameters $X$, the 
+conditional expectation simplifies:
+
+$$
+\textsf{E}(g(X, Y) \mid X) 
+= \textsf{E}(g(X) \mid X) = g(X).
+$$
+
+Our conditional estimator therefore becomes:
+
+$$
+\frac{1}{M} \sum_{i = 1}^M g(x_i).
+$$
+
+This makes performing inference straightforward. First, we write a Stan program 
+that implements the marginalized posterior $p_X(x)$. Running this model will 
+produce samples from the marginalized posterior. We then evaluate $g(\cdot)$ at
+each sample, and use the average over the samples to estimate the expectation 
+of interest. That is, we use Stan `as usual', except that we implement a
+marginalized posterior.
+
+### Recovering the discrete parameters {-}
+
+For the discrete parameters, we are often only interested in the marginal 
+posterior mass function.  For example, for a univariate $Y$ we want to 
+calculate $\Pr(Y = k)$ for all possible values of $k$; note that this allows 
```

Comment:
We've also used square brackets here and `\textrm{Pr}`.   I assume that's what `\Pr` is doing, but I didn't even realize we could use definitions like this in markdown.